lib/src/rules/unnecessary_lambdas.dart

```diff
@@ -80,16 +88,17 @@ class _Visitor extends SimpleAstVisitor {
 
   void _visitInvocationExpression(
       InvocationExpression node, FunctionExpression nodeToLint) {
-    if (nodeToLint.parameters.length != node.argumentList.length) {
+    if (nodeToLint.parameters.parameters.length !=
+        node.argumentList.arguments.length) {
```

Comment:
What we were comparing before was the string length of the source code that produced the node, so try adding a space in one place or another to make the lengths different when there ought to be a lint.